src/FSharpVSPowerTools.Core/DepthParser.fs

```diff
@@ -326,20 +326,11 @@ module internal DepthParsing =
 
 open Microsoft.FSharp.Compiler.SourceCodeServices
 open DepthParsing
-open System
-
-type DepthParser private () =
-    let checker = lazy (FSharpChecker.Create())
-
-    member internal __.Checker = checker.Value
-
-    static member internal Instance = DepthParser()
 
+type DepthParser (checker: FSharpChecker) =
     /////////////////////////////////////////////////////
-
     // TODO, consider perf, way just to consider viewport and do as little work as necessary?
-
-    member internal x.GetRangesImpl(sourceCodeLinesOfTheFile, sourceCodeOfTheFile, filename, checker: FSharpChecker option) =
+    let getRangesImpl(sourceCodeLinesOfTheFile, sourceCodeOfTheFile, filename) =
```

Comment:
Make this function static let and pass the checker explicitly (do similarly for other static members). I think it makes the intention clear and it is consistent with XmlDocParser.